### PR TITLE
Adding a Docker build only template

### PR DESCRIPTION
**Pull request by rakol-ms**
Signed-off-by: Rajendra Kolli <rakol@microsoft.com>

### CODE REVIEW

File: templates/docker-build.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    pool:
+      vmImage: $(vmImageName)
+    steps:
+    - task: Docker@2
+      displayName: Build an image
+      inputs:
+        command: build
+        dockerfile: $(dockerfilePath)
+        tags: |
+          $(tag)
```
Comment (ds-ms) [inline templates/docker-build.yml]: I think the default tag itself is $(Build.BuildId); you don't need to add it explicitly maybe?

File: templates/docker-build.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  - job: Build
+    displayName: Build
+    pool:
+      vmImage: $(vmImageName)
+    steps:
+    - task: Docker@2
+      displayName: Build an image
+      inputs:
+        command: build
+        dockerfile: $(dockerfilePath)
```
Comment (ds-ms) [inline templates/docker-build.yml]: Given that this is being used only once, there's no need for the variable maybe?

File: templates/docker-build.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  vmImageName: 'ubuntu-latest'
+
+stages:
+- stage: Build
+  displayName: Build image stage
+  jobs:  
+  - job: Build
+    displayName: Build
+    pool:
+      vmImage: $(vmImageName)
```
Comment (ds-ms) [inline templates/docker-build.yml]: Given that it's a single stage, you can hard-code it here?

File: templates/docker-build.yml (hunk 1 above)
Comment (rakol-ms) [reply templates/docker-build.yml]: Done. Removed

File: templates/docker-build.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+trigger:
+- master
+
+resources:
+- repo: self
+
+stages:
+- stage: Build
+  displayName: Build image stage
```
Comment (ds-ms) [inline templates/docker-build.yml]: `Build image` should do

File: templates/docker-build.yml (hunk 1 above)
Comment (shashankbarsin) [reply templates/docker-build.yml]: No, let it be added explicitly. It'll be clearer for first time users reading the pipeline definition to understand stuff.

File: templates/docker-build.yml (hunk 4 above)
Comment (shashankbarsin) [reply templates/docker-build.yml]: I agree

**Pull request merged**